torch/distributed/_sharded_tensor/ops/embedding_bag.py

```diff
@@ -367,27 +485,22 @@ def _handle_row_wise_sharding(
     # For Mean, we cannot do the division until very end because the sum of means
     # not equal to the mean of sum. (Divisor is different)
     if mode == "mean":
-        # For a 2D tensor, we just average by col number
```

Comment:
Do we no longer need this handling for 2D tensor?

Reply:
Oh because I now merge the logic of 1D and 2D so that we don't need to treat them separately in the end.  So padding makes a 2D matrix similar like 1D because we can not directly use the col num as the divisor. So we created a new variable `split_sizes_1d_with_padding` to be the divisor.